### ISSUE TITLE
Telegraf PostgreSQL plugin crashes

### ISSUE THREAD

**Issue by mhoshi-vm** [wavefrontHQ/wavefront-collector-for-kubernetes#99]
Currently when enabling the postgresql telegraf plugin (using runtime configs) leads to crashing.

wavefront-collector version 1.2.5

**Steps:**

Run a simple postgresql

```
helm install tac/postgresql-ha --generate-name --version 6.6.4
```

Apply the following Yaml (password may differ)

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: postgresql-collector
  namespace: tanzu-observability-saas
  annotations:
    wavefront.com/discovery-config: 'true'
data:
  collector.yaml: |
    plugins:
    - name: postgresql
      type: telegraf/postgresql
      selectors:
        labels:
          app.kubernetes.io/component:
          - postgresql
        namespaces:
        - default
      port: 5432
      collection:
        interval: "10s"
      conf: |
        address = "postgres://postgres:e8CZuc0UqF@${host}:${port}/postgres"
```

**Result**

Wavefront collector crashes with the following message.

```
time="2021-03-11T00:05:49Z" level=info msg="Adding provider" collection_interval=10s name="telegraf_provider: default-pod-postgresql-ha-1615337944-postgresql-1:5432" timeout=30s
time="2021-03-11T00:05:49Z" level=info msg="Adding provider" collection_interval=3m0s name="telegraf_provider: default-pod-petclinic-db-mariadb-0:3306" timeout=30s
time="2021-03-11T00:05:49Z" level=info msg="Adding provider" collection_interval=10s name="telegraf_provider: default-pod-postgresql-ha-1615337944-postgresql-0:5432" timeout=30s
time="2021-03-11T00:05:59Z" level=info msg="Querying source" name=telegraf_postgresql_plugin_source
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x13475d1]

goroutine 235 [running]:
sync.(*Mutex).Lock(...)
	/usr/local/go/src/sync/mutex.go:74
database/sql.(*DB).conn(0x0, 0x1c7fe00, 0xc0000be058, 0x1, 0x10000c0001df8b8, 0x0, 0x40)
	/usr/local/go/src/database/sql/sql.go:1134 +0x41
database/sql.(*DB).query(0x0, 0x1c7fe00, 0xc0000be058, 0xc000314800, 0x40, 0x0, 0x0, 0x0, 0xc00043d001, 0x40, ...)
	/usr/local/go/src/database/sql/sql.go:1552 +0x66
database/sql.(*DB).QueryContext(0x0, 0x1c7fe00, 0xc0000be058, 0xc000314800, 0x40, 0x0, 0x0, 0x0, 0x4cde38, 0x0, ...)
	/usr/local/go/src/database/sql/sql.go:1534 +0xd1
database/sql.(*DB).Query(...)
	/usr/local/go/src/database/sql/sql.go:1548
github.com/wavefronthq/wavefront-collector-for-kubernetes/vendor/github.com/influxdata/telegraf/plugins/inputs/postgresql.(*Postgresql).Gather(0xc000696880, 0x1c98ea0, 0xc0004d2940, 0x0, 0x0)
	/go/src/github.com/wavefronthq/wavefront-collector-for-kubernetes/vendor/github.com/influxdata/telegraf/plugins/inputs/postgresql/postgresql.go:87 +0xae
github.com/wavefronthq/wavefront-collector-for-kubernetes/plugins/sources/telegraf.(*telegrafPluginSource).ScrapeMetrics(0xc0002f4000, 0x4, 0xc000327d88, 0x1)
	/go/src/github.com/wavefronthq/wavefront-collector-for-kubernetes/plugins/sources/telegraf/provider.go:101 +0xca
github.com/wavefronthq/wavefront-collector-for-kubernetes/plugins/sources.scrape(0x1c801c0, 0xc000493c00, 0xc00009c9c0)
	/go/src/github.com/wavefronthq/wavefront-collector-for-kubernetes/plugins/sources/manager.go:234 +0x200
github.com/wavefronthq/wavefront-collector-for-kubernetes/plugins/sources.(*sourceManagerImpl).AddProvider.func1(0xc000673720, 0x1c801c0, 0xc000493c00, 0xc0004488c0, 0xc000387a40)
	/go/src/github.com/wavefronthq/wavefront-collector-for-kubernetes/plugins/sources/manager.go:160 +0x5f
created by github.com/wavefronthq/wavefront-collector-for-kubernetes/plugins/sources.(*sourceManagerImpl).AddProvider
	/go/src/github.com/wavefronthq/wavefront-collector-for-kubernetes/plugins/sources/manager.go:156 +0x581
```

**Comment by mhoshi-vm** [comment]
Maybe related to this bug https://github.com/influxdata/telegraf/issues/6712 ?

**Comment by josephgee** [comment]
Syncing with K8SSAAS-436

**Comment by priyaselvaganesan** [comment]
@mhoshi-vm, This issue seems to be resolved with [PR #248's](https://github.com/wavefrontHQ/wavefront-collector-for-kubernetes/pull/248) fix. If you still observe the problem, can you please re-open the issue.
